runtime/gc_base/IndexableObjectAllocationModel.cpp

```diff
@@ -287,5 +316,78 @@ MM_IndexableObjectAllocationModel::layoutDiscontiguousArraylet(MM_EnvironmentBas
 	return spine;
 }
 
+#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
+#if !defined(LINUX) || !defined(J9VM_ENV_DATA64) || defined(OMRZTPF)
+/* Double map is only supported on LINUX 64 bit Systems for now */
+#error "Platform not supported by Double Map API"
+#endif /* !definedLINUX || !defined(J9VM_ENV_DATA64) || defined(OMRZTPF) */
+void * 
+MM_IndexableObjectAllocationModel::doubleMapArraylets(MM_EnvironmentBase *env, J9Object *objectPtr) 
+{
+	MM_GCExtensions *extensions = MM_GCExtensions::getExtensions(env);
+	J9JavaVM *javaVM = extensions->getJavaVM();
+	PORT_ACCESS_FROM_ENVIRONMENT(env);
+
+	GC_ArrayletLeafIterator arrayletLeafIterator(javaVM, (J9IndexableObject *)objectPtr);
+	UDATA arrayletLeafCount = arrayletLeafIterator.getNumLeafs();
+	MM_Heap *heap = extensions->getHeap();
+	UDATA arrayletLeafSize = env->getOmrVM()->_arrayletLeafSize;
+
+	/* Adjust arraylet leaf count. There might be the case where the last leaf is NULL */
+	if (_dataSize % arrayletLeafSize == 0) {
+		arrayletLeafCount -= 1;
+	}
```

Comment:
This rely on fact extra dummy leaf pointer added. When we remove it this code would be incorrect but might be missed to fix. Can it be more general here like:
`arrayletLeafCount = MM_Math::roundToCeiling(arrayletLeafSize, _dataSize)  / arrayletLeafSize;`